compiler/analyzer.ml

```diff
@@ -283,14 +304,9 @@ and check_binop env e1 op e2 =
         env', Sast.Expr(Sast.Binop(ew1', op, ew2'), result_type)
       else binop_error typ1 op typ2
 
-    (* Equality operations - overloaded, no constraining can be done *)
-    | Eq | Neq -> 
-      let is_valid_equality = function
-        | Num | Bool | String | Unconst -> true
-        | _ -> false in 
-      if is_valid_equality typ1 && is_valid_equality typ2 then 
-        env', Sast.Expr(Sast.Binop(ew1, op, ew2), Bool)
-      else binop_error typ1 op typ2
+    (* Equality operations - overloaded, no constraining can be done, can take
```

Comment:
ahh so i meant to tell you guys about this: i think every single data type in our lang should evaluate to a bool because that's the way it is in python